superset-frontend/src/explore/components/controls/DndColumnSelectControl/ColumnSelectPopover.tsx

```diff
@@ -283,16 +290,72 @@ const ColumnSelectPopover = ({
                     )
               }
             />
+          ) : (
+            <EmptyStateSmall
+              image="empty.svg"
+              title={
+                isTemporal
+                  ? t('No temporal columns found')
+                  : t('No saved expressions found')
+              }
+              description={
+                isTemporal ? (
+                  <>
+                    <span
+                      role="button"
+                      tabIndex={0}
+                      onClick={() => {
+                        if (setDatasetModal) setDatasetModal(true);
+                        onClose();
+                      }}
+                    >
+                      {t('Create a dataset')}
+                    </span>{' '}
+                    {t(' to mark a column as a time column')}
+                  </>
+                ) : (
+                  <>
+                    <span
+                      role="button"
+                      tabIndex={0}
+                      onClick={() => {
+                        if (setDatasetModal) setDatasetModal(true);
+                        onClose();
+                      }}
+                    >
+                      {t('Create a dataset')}
+                    </span>{' '}
+                    {t(' to mark a column as a calculated column')}
+                  </>
+                )
+              }
+            />
           )}
         </Tabs.TabPane>
         <Tabs.TabPane key="simple" tab={t('Simple')}>
           {isTemporal && simpleColumns.length === 0 ? (
             <EmptyStateSmall
               image="empty.svg"
               title={t('No temporal columns found')}
-              description={t(
-                'Mark a column as temporal in "Edit datasource" modal',
-              )}
+              description={
+                datasource.type === 'dataset' ? (
+                  t('Mark a column as temporal in "Edit datasource" modal')
+                ) : (
+                  <>
+                    <span
+                      role="button"
+                      tabIndex={0}
+                      onClick={() => {
+                        if (setDatasetModal) setDatasetModal(true);
+                        onClose();
```

Comment:
It looks like we're using this same onclick function three times. Do you think we can dry this up?